cmd/influxd/restore/restore.go

```diff
@@ -227,41 +433,153 @@ func (cmd *Command) unpackShard(shardID string) error {
 	}
 
 	// find the shard backup files
-	pat := filepath.Join(cmd.backupFilesPath, fmt.Sprintf(backup_util.BackupFilePattern, cmd.database, cmd.retention, id))
+	pat := filepath.Join(cmd.backupFilesPath, fmt.Sprintf(backup_util.BackupFilePattern, cmd.destinationDatabase, cmd.restoreRetention, id))
 	return cmd.unpackFiles(pat + ".*")
 }
 
-// unpackDatabase will look for all backup files in the path matching this database
+// unpackFiles will look for backup files matching the pattern and restore them to the data dir
+func (cmd *Command) uploadShardsLive() error {
+	// find the destinationDatabase backup files
+	if cmd.enterprise {
+		for _, file := range cmd.manifest.Files {
+			if cmd.sourceDatabase == "" || cmd.sourceDatabase == file.Database {
+				if cmd.backupRetention == "" || cmd.backupRetention == file.Policy {
+					if cmd.shard == "" || cmd.shard == strconv.FormatUint(file.ShardID, 10) {
+						cmd.StdoutLogger.Printf("Restoring shard %d live from backup %s\n", file.ShardID, file.FileName)
+						f, err := os.Open(filepath.Join(cmd.backupFilesPath, file.FileName))
+						if err != nil {
+							return err
+						}
+						gr, err := gzip.NewReader(f)
+						if err != nil {
+							return err
+						}
+						tr := tar.NewReader(gr)
+						if err := cmd.uploadShardLive(file.ShardID, tr); err != nil {
+							return err
+						}
+					}
+				}
+			}
+		}
+	} else {
+		pat := fmt.Sprintf("%s.*", filepath.Join(cmd.backupFilesPath, cmd.sourceDatabase))
+		cmd.StdoutLogger.Printf("Restoring live from backup %s\n", pat)
+		backupFiles, err := filepath.Glob(pat)
+		if err != nil {
+			return err
+		}
+		if len(backupFiles) == 0 {
+			return fmt.Errorf("no backup files in %s", cmd.backupFilesPath)
+		}
+
+		for _, fn := range backupFiles {
+			parts := strings.Split(fn, ".")
+
+			if len(parts) != 4 {
+				cmd.StderrLogger.Printf("Skipping mis-named backup file: %s", fn)
+			}
+			shardID, err := strconv.ParseUint(parts[2], 10, 64)
+			if err != nil {
+				return err
+			}
+			f, err := os.Open(fn)
+			if err != nil {
+				return err
+			}
+			tr := tar.NewReader(f)
+			if err := cmd.uploadShardLive(shardID, tr); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (cmd *Command) uploadShardLive(shardID uint64, tr *tar.Reader) error {
+	newShardID := cmd.shardIDMap[shardID]
+
+	conn, err := tcp.Dial("tcp", cmd.host, snapshotter.MuxHeader)
+	defer conn.Close()
+	if err != nil {
+		return err
+	}
+
+	if _, err := conn.Write([]byte{byte(snapshotter.RequestShardUpdate)}); err != nil {
+		return err
+	}
+
+	// 0.  write the shard ID to pw
```

Comment:
What is `pw`? I also feel like this code should be abstracted away, maybe in the `snapshotter` package, so that clients don't have to know the exact byte order and incantations to perform operations.